src/components/organisms/MarketInfoDialog/MarketInfoDialog.tsx

```diff
@@ -0,0 +1,149 @@
+import { Dialog } from '@headlessui/react';
+import { XMarkIcon } from '@heroicons/react/20/solid';
+import { formatDate } from '@secured-finance/sf-core';
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+import { useEffect, useState } from 'react';
+import ArrowUpSquare from 'src/assets/icons/arrow-up-square.svg';
+import DocumentTextIcon from 'src/assets/icons/document-text.svg';
+import { Tooltip } from 'src/components/molecules';
+import {
+    CountdownFormat,
+    currencyMap,
+    CurrencySymbol,
+    formatLoanValue,
+    getCountdown,
+    handlePriceSource,
+} from 'src/utils';
+import { MarketInfoDialogProps } from './types';
+
+dayjs.extend(duration);
+
+export const MarketInfoDialog = ({
+    isOpen,
+    onClose,
+    currency,
+    currentMarket,
+    currencyPrice,
+    marketInfo,
+    lastLoanValue,
+}: MarketInfoDialogProps) => {
+    const maturity = currentMarket?.value.maturity ?? 0;
+    const targetTime = maturity * 1000;
+    const [time, setTime] = useState<CountdownFormat | undefined>(
+        getCountdown(targetTime)
+    );
+    const lastPrice = formatLoanValue(currentMarket?.value, 'price');
+    const CurrencyIcon = currencyMap[currency as CurrencySymbol]?.icon;
+
+    const priceSource = handlePriceSource(currency as CurrencySymbol);
+
+    useEffect(() => {
```

Comment:
this useeffect can be a hook in itself.
I think this is now being used at 3 places in the code and twice in this PR

Reply:
added a useGetCountdown took to handle that